server/src/core/user/user-service.ts

```diff
@@ -104,37 +113,44 @@ export class Users implements UserService {
     const user: DbUser = {
       _id: generateId(),
       phone: args.phone,
+      name: args.name,
       addedBy: '',
       donors: [],
       roles: ['donor'],
       createdAt: now,
       updatedAt: now
     };
+
     try {
-      if (args.password) 
+      if (args.password) {
         user.password = await hashPassword(args.password);
-      else if (args.googleIdToken)
-        user.email = await verifyGoogleIdToken(args.googleIdToken);
+      }
+      else if (args.googleIdToken) {
+        const googleUserData = await verifyGoogleIdToken(args.googleIdToken);
+        user.email = googleUserData.email;
+        user.name = googleUserData.name;
+      }
 
       const res = await this.collection.insertOne(user);
       return getSafeUser(res.ops[0]);
     }
     catch (e) {
-      if (e instanceof AppError) throw e;
+      rethrowIfAppError(e);
+
       if (isMongoDuplicateKeyError(e, args.phone)) {
         const existingUser = await this.collection.findOne({ phone: args.phone });
         if (existingUser.email && user.email && existingUser.email === user.email)
           return getSafeUser(existingUser)
         throw createUniquenessFailedError(messages.ERROR_PHONE_ALREADY_IN_USE);
       }
+
       throw createDbOpFailedError(e.message);
     }
   }
 
   async nominateBeneficiary(args: UserNominateArgs): Promise<User> {
-    const { phone, email, nominator } = args;
-    if (email) validators.validatesNominate({ phone, email, nominator });
```

Comment:
You shouldn't have removed these `if` `else` checks. They were there to ensure that if no email was provided by the nominator, we do not pass a null value to the validator, cause then what happens is the validator complains that email cannot be anything other than a string.

Reply:
If the email is not provided, then why would it be `null`? `null` means the email is provided and set to `null`.

Reply:
I had explicitly set it to null to make it easier to detect during validation that no email was provided.

Reply:
@BambanzaJuniorThe2nd I think not providing an email is a better way to not provide an email :)

Reply:
The app also doesn't throw any errors when you sign up without an email. So I don't see what benefit adding an if-check provides.

Reply:
I noticed that you get a validation error if you nominate a user when you leave the email field empty on the frontend form. I've fixed that by omitting the email field from the request payload when the email field is empty.